client/core/account.go

```diff
@@ -2,11 +2,69 @@ package core
 
 import (
 	"encoding/hex"
+	"fmt"
 
 	"decred.org/dcrdex/client/db"
 	"github.com/decred/dcrd/dcrec/secp256k1/v3"
 )
 
+// AccountDisable is used to disable an account by given host and application
+// password.
+func (c *Core) AccountDisable(pw []byte, host string) error {
+	// Validate password.
+	_, err := c.encryptionKey(pw)
+	if err != nil {
+		return codedError(passwordErr, err)
+	}
+
+	// Parse address.
+	addr, err := addrHost(host)
+	if err != nil {
+		return newError(addressParseErr, "error parsing address: %v", err)
+	}
+
+	// Get the dexConnection.
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+	dc, found := c.conns[addr]
+	if !found {
+		return newError(unknownDEXErr, "DEX: %s", addr)
+	}
+
+	// Check active orders.
+	if dc.hasActiveOrders() {
+		return fmt.Errorf("cannot disable account with active orders")
+	}
+
+	acctInfo, err := c.db.Account(dc.acct.host)
+	if err != nil {
+		return newError(accountRetrieveErr, "Error retrieving account: %v", err)
+	}
+	err = c.db.DisableAccount(acctInfo)
+	if err != nil {
+		return newError(accountDisableErr, "Error disabling account: %v", err)
+	}
+	// Stop dexConnection books.
+	dc.cfgMtx.RLock()
+	for _, m := range dc.cfg.Markets {
+		// Emptry bookie's feeds map & stop close timers.
+		dc.booksMtx.Lock()
+		if b, found := dc.books[m.Name]; found {
+			b.feeds = make(map[uint32]*BookFeed, 1)
+			b.closeTimer.Stop()
+		}
+		dc.booksMtx.Unlock()
```

Comment:
`booksMtx` is for `dc.books`. `b.mtx` is for the `bookie` fields.

Reply:
I'm iterating on dc.books here no ?

Reply:
You're reading and writing to `bookie` fields too

Reply:
Oh I see